Clean build on FC24 and LLVM 3.7.1/3.8.1 fails with linker errors

I'm in the process of creating a clean environment for bcc hands-on labs, and can't get bcc (master) to build. Followed the INSTALL.md instructions, tried both LLVM 3.7.1 (as in the instructions) and 3.8.1 and getting a mass of linker errors.

Has there been an ABI change somewhere down the road? Or is this simply an incompatibility with FC24 and I should use FC23?

```
../../src/cc/libbcc.a(bpf_module.cc.o): In function `ebpf::BPFModule::dump_ir(llvm::Module&)':
bpf_module.cc:(.text+0x4b30): undefined reference to `llvm::createPrintModulePass(llvm::raw_ostream&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
../../src/cc/libbcc.a(bpf_module.cc.o): In function `ebpf::BPFModule::run_pass_manager(llvm::Module&)':
bpf_module.cc:(.text+0x4c5d): undefined reference to `llvm::createPrintModulePass(llvm::raw_ostream&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
../../src/cc/libbcc.a(bpf_module.cc.o): In function `llvm::RTDyldMemoryManager::getSymbolAddress(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
bpf_module.cc:(.text._ZN4llvm19RTDyldMemoryManager16getSymbolAddressERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN4llvm19RTDyldMemoryManager16getSymbolAddressERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x4): undefined reference to `llvm::RTDyldMemoryManager::getSymbolAddressInProcess(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
../../src/cc/libbcc.a(bpf_module.cc.o): In function `llvm::RTDyldMemoryManager::findSymbol(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
bpf_module.cc:(.text._ZN4llvm19RTDyldMemoryManager10findSymbolERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN4llvm19RTDyldMemoryManager10findSymbolERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x18): undefined reference to `llvm::RTDyldMemoryManager::getSymbolAddressInProcess(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
../../src/cc/libbcc.a(bpf_module.cc.o):(.data.rel.ro._ZTVN4ebpf15MyMemoryManagerE[_ZTVN4ebpf15MyMemoryManagerE]+0x50): undefined reference to `llvm::SectionMemoryManager::finalizeMemory(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
../../src/cc/libbcc.a(bpf_module.cc.o):(.data.rel.ro._ZTVN4ebpf15MyMemoryManagerE[_ZTVN4ebpf15MyMemoryManagerE]+0x88): undefined reference to `llvm::RTDyldMemoryManager::getPointerToNamedFunction(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool)'
/usr/local/lib/libLLVMipo.a(Inliner.cpp.o): In function `llvm::Inliner::getAnalysisUsage(llvm::AnalysisUsage&) const':
/home/nikola/final/llvm.src/lib/Transforms/IPO/Inliner.cpp:(.text._ZNK4llvm7Inliner16getAnalysisUsageERNS_13AnalysisUsageE+0x36): undefined reference to `llvm::CallGraphSCCPass::getAnalysisUsage(llvm::AnalysisUsage&) const'
/usr/local/lib/libLLVMipo.a(Inliner.cpp.o): In function `llvm::Inliner::runOnSCC(llvm::CallGraphSCC&)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/Inliner.cpp:(.text._ZN4llvm7Inliner8runOnSCCERNS_12CallGraphSCCE+0x28): undefined reference to `llvm::CallGraphWrapperPass::ID'
/home/nikola/final/llvm.src/lib/Transforms/IPO/Inliner.cpp:(.text._ZN4llvm7Inliner8runOnSCCERNS_12CallGraphSCCE+0x4c): undefined reference to `llvm::CallGraphWrapperPass::ID'
/home/nikola/final/llvm.src/lib/Transforms/IPO/Inliner.cpp:(.text._ZN4llvm7Inliner8runOnSCCERNS_12CallGraphSCCE+0x6e9): undefined reference to `llvm::CallGraphNode::removeCallEdgeFor(llvm::CallSite)'
/home/nikola/final/llvm.src/lib/Transforms/IPO/Inliner.cpp:(.text._ZN4llvm7Inliner8runOnSCCERNS_12CallGraphSCCE+0xdee): undefined reference to `llvm::CallGraph::removeFunctionFromModule(llvm::CallGraphNode*)'
/usr/local/lib/libLLVMipo.a(Inliner.cpp.o): In function `llvm::Inliner::removeDeadFunctions(llvm::CallGraph&, bool)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/Inliner.cpp:(.text._ZN4llvm7Inliner19removeDeadFunctionsERNS_9CallGraphEb+0x272): undefined reference to `llvm::CallGraphNode::removeAnyCallEdgeTo(llvm::CallGraphNode*)'
/home/nikola/final/llvm.src/lib/Transforms/IPO/Inliner.cpp:(.text._ZN4llvm7Inliner19removeDeadFunctionsERNS_9CallGraphEb+0x832): undefined reference to `llvm::CallGraphNode::removeAnyCallEdgeTo(llvm::CallGraphNode*)'
/home/nikola/final/llvm.src/lib/Transforms/IPO/Inliner.cpp:(.text._ZN4llvm7Inliner19removeDeadFunctionsERNS_9CallGraphEb+0x947): undefined reference to `llvm::CallGraph::removeFunctionFromModule(llvm::CallGraphNode*)'
/usr/local/lib/libLLVMipo.a(Inliner.cpp.o):(.data.rel.ro._ZTVN4llvm7InlinerE+0x40): undefined reference to `llvm::CallGraphSCCPass::createPrinterPass(llvm::raw_ostream&, std::string const&) const'
/usr/local/lib/libLLVMipo.a(Inliner.cpp.o):(.data.rel.ro._ZTVN4llvm7InlinerE+0x48): undefined reference to `llvm::CallGraphSCCPass::assignPassManager(llvm::PMStack&, llvm::PassManagerType)'
/usr/local/lib/libLLVMipo.a(Internalize.cpp.o): In function `(anonymous namespace)::InternalizePass::getAnalysisUsage(llvm::AnalysisUsage&) const':
/home/nikola/final/llvm.src/lib/Transforms/IPO/Internalize.cpp:(.text._ZNK12_GLOBAL__N_115InternalizePass16getAnalysisUsageERN4llvm13AnalysisUsageE+0x1f): undefined reference to `llvm::CallGraphWrapperPass::ID'
/usr/local/lib/libLLVMipo.a(Internalize.cpp.o): In function `(anonymous namespace)::InternalizePass::runOnModule(llvm::Module&)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/Internalize.cpp:(.text._ZN12_GLOBAL__N_115InternalizePass11runOnModuleERN4llvm6ModuleE+0x1e): undefined reference to `llvm::CallGraphWrapperPass::ID'
/home/nikola/final/llvm.src/lib/Transforms/IPO/Internalize.cpp:(.text._ZN12_GLOBAL__N_115InternalizePass11runOnModuleERN4llvm6ModuleE+0x3e): undefined reference to `llvm::CallGraphWrapperPass::ID'
/home/nikola/final/llvm.src/lib/Transforms/IPO/Internalize.cpp:(.text._ZN12_GLOBAL__N_115InternalizePass11runOnModuleERN4llvm6ModuleE+0x311): undefined reference to `llvm::CallGraphNode::removeOneAbstractEdgeTo(llvm::CallGraphNode*)'
/usr/local/lib/libLLVMipo.a(PassManagerBuilder.cpp.o): In function `llvm::PassManagerBuilder::addLTOOptimizationPasses(llvm::legacy::PassManagerBase&)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/PassManagerBuilder.cpp:(.text._ZN4llvm18PassManagerBuilder24addLTOOptimizationPassesERNS_6legacy15PassManagerBaseE+0x1ff): undefined reference to `llvm::createGlobalsModRefPass()'
/usr/local/lib/libLLVMTransformUtils.a(InlineFunction.cpp.o): In function `llvm::InlineFunction(llvm::CallSite, llvm::InlineFunctionInfo&, bool)':
/home/nikola/final/llvm.src/lib/Transforms/Utils/InlineFunction.cpp:(.text._ZN4llvm14InlineFunctionENS_8CallSiteERNS_18InlineFunctionInfoEb+0x121b): undefined reference to `llvm::CallGraphNode::removeCallEdgeFor(llvm::CallSite)'
../../src/cc/frontends/clang/libclang_frontend.a(b_frontend_action.cc.o): In function `ebpf::BTypeVisitor::VisitBinaryOperator(clang::BinaryOperator*)':
b_frontend_action.cc:(.text+0x10db): undefined reference to `clang::Rewriter::getRewrittenText[abi:cxx11](clang::SourceRange) const'
b_frontend_action.cc:(.text+0x110e): undefined reference to `clang::Rewriter::getRewrittenText[abi:cxx11](clang::SourceRange) const'
../../src/cc/frontends/clang/libclang_frontend.a(b_frontend_action.cc.o): In function `ebpf::BTypeVisitor::VisitFunctionDecl(clang::FunctionDecl*)':
b_frontend_action.cc:(.text+0x2115): undefined reference to `clang::Rewriter::getRewrittenText[abi:cxx11](clang::SourceRange) const'
../../src/cc/frontends/clang/libclang_frontend.a(b_frontend_action.cc.o): In function `ebpf::BTypeVisitor::VisitCallExpr(clang::CallExpr*)':
b_frontend_action.cc:(.text+0x335d): undefined reference to `clang::Rewriter::getRewrittenText[abi:cxx11](clang::SourceRange) const'
b_frontend_action.cc:(.text+0x35ed): undefined reference to `clang::Rewriter::getRewrittenText[abi:cxx11](clang::SourceRange) const'
../../src/cc/frontends/clang/libclang_frontend.a(b_frontend_action.cc.o):b_frontend_action.cc:(.text+0x37dd): more undefined references to `clang::Rewriter::getRewrittenText[abi:cxx11](clang::SourceRange) const' follow
../../src/cc/frontends/clang/libclang_frontend.a(b_frontend_action.cc.o): In function `ebpf::ProbeVisitor::VisitMemberExpr(clang::MemberExpr*)':
b_frontend_action.cc:(.text+0x745e): undefined reference to `clang::QualType::getAsString[abi:cxx11](clang::Type const*, clang::Qualifiers)'
b_frontend_action.cc:(.text+0x74e0): undefined reference to `clang::QualType::getAsString[abi:cxx11](clang::Type const*, clang::Qualifiers)'
../../src/cc/frontends/clang/libclang_frontend.a(b_frontend_action.cc.o): In function `ebpf::ProbeVisitor::VisitUnaryOperator(clang::UnaryOperator*)':
b_frontend_action.cc:(.text+0x7a98): undefined reference to `clang::Rewriter::getRewrittenText[abi:cxx11](clang::SourceRange) const'
b_frontend_action.cc:(.text+0x7adf): undefined reference to `clang::QualType::getAsString[abi:cxx11](clang::Type const*, clang::Qualifiers)'
../../src/cc/frontends/clang/libclang_frontend.a(b_frontend_action.cc.o): In function `ebpf::BMapDeclVisitor::VisitRecordDecl(clang::RecordDecl*)':
b_frontend_action.cc:(.text+0x812c): undefined reference to `llvm::APInt::toString[abi:cxx11](unsigned int, bool) const'
../../src/cc/frontends/clang/libclang_frontend.a(tp_frontend_action.cc.o): In function `ebpf::TracepointTypeVisitor::VisitFunctionDecl(clang::FunctionDecl*)':
tp_frontend_action.cc:(.text+0x16ff): undefined reference to `clang::QualType::getAsString[abi:cxx11](clang::Type const*, clang::Qualifiers)'
/usr/local/lib/libLLVMCoverage.a(CoverageMapping.cpp.o): In function `llvm::coverage::CounterMappingContext::evaluate(llvm::coverage::Counter const&) const':
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZNK4llvm8coverage21CounterMappingContext8evaluateERKNS0_7CounterE+0xd9): undefined reference to `llvm::errorCodeToError(std::error_code)'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZNK4llvm8coverage21CounterMappingContext8evaluateERKNS0_7CounterE+0x108): undefined reference to `llvm::errorCodeToError(std::error_code)'
/usr/local/lib/libLLVMCoverage.a(CoverageMapping.cpp.o): In function `llvm::coverage::CoverageMapping::load(llvm::coverage::CoverageMappingReader&, llvm::IndexedInstrProfReader&)':
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm8coverage15CoverageMapping4loadERNS0_21CoverageMappingReaderERNS_22IndexedInstrProfReaderE+0xa9): undefined reference to `vtable for llvm::InstrProfError'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm8coverage15CoverageMapping4loadERNS0_21CoverageMappingReaderERNS_22IndexedInstrProfReaderE+0x28d): undefined reference to `llvm::getFuncNameWithoutPrefix(llvm::StringRef, llvm::StringRef)'
/usr/local/lib/libLLVMCoverage.a(CoverageMapping.cpp.o): In function `llvm::coverage::CoverageMapping::load(llvm::StringRef, llvm::StringRef, llvm::StringRef)':
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm8coverage15CoverageMapping4loadENS_9StringRefES2_S2_+0x4e): undefined reference to `llvm::MemoryBuffer::getFileOrSTDIN(llvm::Twine const&, long, bool)'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm8coverage15CoverageMapping4loadENS_9StringRefES2_S2_+0x6c): undefined reference to `llvm::errorCodeToError(std::error_code)'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm8coverage15CoverageMapping4loadENS_9StringRefES2_S2_+0xfd): undefined reference to `llvm::IndexedInstrProfReader::create(llvm::Twine const&)'
/usr/local/lib/libLLVMCoverage.a(CoverageMapping.cpp.o): In function `llvm::ErrorInfo<llvm::coverage::CoverageMapError, llvm::ErrorInfoBase>::isA(void const*) const':
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZNK4llvm9ErrorInfoINS_8coverage16CoverageMapErrorENS_13ErrorInfoBaseEE3isAEPKv[_ZNK4llvm9ErrorInfoINS_8coverage16CoverageMapErrorENS_13ErrorInfoBaseEE3isAEPKv]+0xd): undefined reference to `llvm::ErrorInfoBase::ID'
/usr/local/lib/libLLVMCoverage.a(CoverageMapping.cpp.o): In function `llvm::Error llvm::handleErrors<llvm::consumeError(llvm::Error)::{lambda(llvm::ErrorInfoBase const&)#1}>(llvm::Error, llvm::consumeError(llvm::Error)::{lambda(llvm::ErrorInfoBase const&)#1}&&)':
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm12handleErrorsIJZNS_12consumeErrorENS_5ErrorEEUlRKNS_13ErrorInfoBaseEE_EEES1_S1_DpOT_[_ZN4llvm12handleErrorsIJZNS_12consumeErrorENS_5ErrorEEUlRKNS_13ErrorInfoBaseEE_EEES1_S1_DpOT_]+0x2a): undefined reference to `llvm::ErrorList::ID'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm12handleErrorsIJZNS_12consumeErrorENS_5ErrorEEUlRKNS_13ErrorInfoBaseEE_EEES1_S1_DpOT_[_ZN4llvm12handleErrorsIJZNS_12consumeErrorENS_5ErrorEEUlRKNS_13ErrorInfoBaseEE_EEES1_S1_DpOT_]+0x81): undefined reference to `llvm::ErrorInfoBase::ID'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm12handleErrorsIJZNS_12consumeErrorENS_5ErrorEEUlRKNS_13ErrorInfoBaseEE_EEES1_S1_DpOT_[_ZN4llvm12handleErrorsIJZNS_12consumeErrorENS_5ErrorEEUlRKNS_13ErrorInfoBaseEE_EEES1_S1_DpOT_]+0x11a): undefined reference to `llvm::ErrorInfoBase::ID'
/usr/local/lib/libLLVMCoverage.a(CoverageMapping.cpp.o): In function `llvm::ErrorList::join(llvm::Error, llvm::Error)':
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm9ErrorList4joinENS_5ErrorES1_[_ZN4llvm9ErrorList4joinENS_5ErrorES1_]+0x34): undefined reference to `llvm::ErrorList::ID'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm9ErrorList4joinENS_5ErrorES1_[_ZN4llvm9ErrorList4joinENS_5ErrorES1_]+0x59): undefined reference to `llvm::ErrorList::ID'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm9ErrorList4joinENS_5ErrorES1_[_ZN4llvm9ErrorList4joinENS_5ErrorES1_]+0xfb): undefined reference to `llvm::ErrorList::ID'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm9ErrorList4joinENS_5ErrorES1_[_ZN4llvm9ErrorList4joinENS_5ErrorES1_]+0x17d): undefined reference to `vtable for llvm::ErrorList'
/usr/local/lib/libLLVMCoverage.a(CoverageMapping.cpp.o): In function `llvm::Error llvm::handleErrors<llvm::InstrProfError::take(llvm::Error)::{lambda(llvm::InstrProfError const&)#1}>(llvm::Error, llvm::InstrProfError::take(llvm::Error)::{lambda(llvm::InstrProfError const&)#1}&&)':
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm12handleErrorsIJZNS_14InstrProfError4takeENS_5ErrorEEUlRKS1_E_EEES2_S2_DpOT_[_ZN4llvm12handleErrorsIJZNS_14InstrProfError4takeENS_5ErrorEEUlRKS1_E_EEES2_S2_DpOT_]+0x2f): undefined reference to `llvm::ErrorList::ID'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm12handleErrorsIJZNS_14InstrProfError4takeENS_5ErrorEEUlRKS1_E_EEES2_S2_DpOT_[_ZN4llvm12handleErrorsIJZNS_14InstrProfError4takeENS_5ErrorEEUlRKS1_E_EEES2_S2_DpOT_]+0x82): undefined reference to `llvm::InstrProfError::ID'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMapping.cpp:(.text._ZN4llvm12handleErrorsIJZNS_14InstrProfError4takeENS_5ErrorEEUlRKS1_E_EEES2_S2_DpOT_[_ZN4llvm12handleErrorsIJZNS_14InstrProfError4takeENS_5ErrorEEUlRKS1_E_EEES2_S2_DpOT_]+0x12d): undefined reference to `llvm::InstrProfError::ID'
/usr/local/lib/libLLVMCoverage.a(CoverageMapping.cpp.o):(.data.rel.ro._ZTVN4llvm8coverage16CoverageMapErrorE+0x40): undefined reference to `llvm::ErrorInfoBase::anchor()'
/usr/local/lib/libLLVMCoverage.a(CoverageMappingReader.cpp.o): In function `llvm::coverage::CoverageMappingIterator::increment()':
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMappingReader.cpp:(.text._ZN4llvm8coverage23CoverageMappingIterator9incrementEv+0x3f): undefined reference to `llvm::ErrorList::ID'
/usr/local/lib/libLLVMCoverage.a(CoverageMappingReader.cpp.o): In function `llvm::InstrProfSymtab::create(llvm::object::SectionRef&)':
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMappingReader.cpp:(.text._ZN4llvm15InstrProfSymtab6createERNS_6object10SectionRefE+0x2b): undefined reference to `llvm::errorCodeToError(std::error_code)'
/usr/local/lib/libLLVMCoverage.a(CoverageMappingReader.cpp.o): In function `llvm::coverage::BinaryCoverageReader::create(std::unique_ptr<llvm::MemoryBuffer, std::default_delete<llvm::MemoryBuffer> >&, llvm::StringRef)':
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMappingReader.cpp:(.text._ZN4llvm8coverage20BinaryCoverageReader6createERSt10unique_ptrINS_12MemoryBufferESt14default_deleteIS3_EENS_9StringRefE+0x4d9): undefined reference to `llvm::errorCodeToError(std::error_code)'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMappingReader.cpp:(.text._ZN4llvm8coverage20BinaryCoverageReader6createERSt10unique_ptrINS_12MemoryBufferESt14default_deleteIS3_EENS_9StringRefE+0x4fd): undefined reference to `llvm::errorCodeToError(std::error_code)'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMappingReader.cpp:(.text._ZN4llvm8coverage20BinaryCoverageReader6createERSt10unique_ptrINS_12MemoryBufferESt14default_deleteIS3_EENS_9StringRefE+0x54e): undefined reference to `llvm::errorCodeToError(std::error_code)'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMappingReader.cpp:(.text._ZN4llvm8coverage20BinaryCoverageReader6createERSt10unique_ptrINS_12MemoryBufferESt14default_deleteIS3_EENS_9StringRefE+0x68b): undefined reference to `llvm::readPGOFuncNameStrings(llvm::StringRef, llvm::InstrProfSymtab&)'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMappingReader.cpp:(.text._ZN4llvm8coverage20BinaryCoverageReader6createERSt10unique_ptrINS_12MemoryBufferESt14default_deleteIS3_EENS_9StringRefE+0x740): undefined reference to `llvm::readPGOFuncNameStrings(llvm::StringRef, llvm::InstrProfSymtab&)'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMappingReader.cpp:(.text._ZN4llvm8coverage20BinaryCoverageReader6createERSt10unique_ptrINS_12MemoryBufferESt14default_deleteIS3_EENS_9StringRefE+0x7ca): undefined reference to `llvm::readPGOFuncNameStrings(llvm::StringRef, llvm::InstrProfSymtab&)'
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMappingReader.cpp:(.text._ZN4llvm8coverage20BinaryCoverageReader6createERSt10unique_ptrINS_12MemoryBufferESt14default_deleteIS3_EENS_9StringRefE+0xa1c): undefined reference to `llvm::readPGOFuncNameStrings(llvm::StringRef, llvm::InstrProfSymtab&)'
/usr/local/lib/libLLVMCoverage.a(CoverageMappingReader.cpp.o): In function `lookupSection(llvm::object::ObjectFile&, llvm::StringRef)':
/home/nikola/final/llvm.src/lib/ProfileData/Coverage/CoverageMappingReader.cpp:(.text._ZL13lookupSectionRN4llvm6object10ObjectFileENS_9StringRefE+0xdf): undefined reference to `llvm::errorCodeToError(std::error_code)'
/usr/local/lib/libLLVMipo.a(InlineAlways.cpp.o): In function `llvm::initializeAlwaysInlinerPass(llvm::PassRegistry&)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/InlineAlways.cpp:(.text._ZN4llvm27initializeAlwaysInlinerPassERNS_12PassRegistryE+0x46): undefined reference to `llvm::initializeCallGraphWrapperPassPass(llvm::PassRegistry&)'
/home/nikola/final/llvm.src/lib/Transforms/IPO/InlineAlways.cpp:(.text._ZN4llvm27initializeAlwaysInlinerPassERNS_12PassRegistryE+0x4e): undefined reference to `llvm::initializeInlineCostAnalysisPass(llvm::PassRegistry&)'
/usr/local/lib/libLLVMipo.a(InlineAlways.cpp.o): In function `(anonymous namespace)::AlwaysInliner::getAnalysisUsage(llvm::AnalysisUsage&) const':
/home/nikola/final/llvm.src/lib/Transforms/IPO/InlineAlways.cpp:(.text._ZNK12_GLOBAL__N_113AlwaysInliner16getAnalysisUsageERN4llvm13AnalysisUsageE+0xd): undefined reference to `llvm::InlineCostAnalysis::ID'
/usr/local/lib/libLLVMipo.a(InlineAlways.cpp.o): In function `(anonymous namespace)::AlwaysInliner::runOnSCC(llvm::CallGraphSCC&)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/InlineAlways.cpp:(.text._ZN12_GLOBAL__N_113AlwaysInliner8runOnSCCERN4llvm12CallGraphSCCE+0x1b): undefined reference to `llvm::InlineCostAnalysis::ID'
/home/nikola/final/llvm.src/lib/Transforms/IPO/InlineAlways.cpp:(.text._ZN12_GLOBAL__N_113AlwaysInliner8runOnSCCERN4llvm12CallGraphSCCE+0x3c): undefined reference to `llvm::InlineCostAnalysis::ID'
/usr/local/lib/libLLVMipo.a(InlineAlways.cpp.o): In function `(anonymous namespace)::AlwaysInliner::getInlineCost(llvm::CallSite)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/InlineAlways.cpp:(.text._ZN12_GLOBAL__N_113AlwaysInliner13getInlineCostEN4llvm8CallSiteE+0xae): undefined reference to `llvm::InlineCostAnalysis::isInlineViable(llvm::Function&)'
/usr/local/lib/libLLVMipo.a(InlineAlways.cpp.o):(.data.rel.ro._ZTVN12_GLOBAL__N_113AlwaysInlinerE+0x40): undefined reference to `llvm::CallGraphSCCPass::createPrinterPass(llvm::raw_ostream&, std::string const&) const'
/usr/local/lib/libLLVMipo.a(InlineAlways.cpp.o):(.data.rel.ro._ZTVN12_GLOBAL__N_113AlwaysInlinerE+0x48): undefined reference to `llvm::CallGraphSCCPass::assignPassManager(llvm::PMStack&, llvm::PassManagerType)'
/usr/local/lib/libLLVMipo.a(InlineSimple.cpp.o): In function `llvm::initializeSimpleInlinerPass(llvm::PassRegistry&)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/InlineSimple.cpp:(.text._ZN4llvm27initializeSimpleInlinerPassERNS_12PassRegistryE+0x46): undefined reference to `llvm::initializeCallGraphWrapperPassPass(llvm::PassRegistry&)'
/home/nikola/final/llvm.src/lib/Transforms/IPO/InlineSimple.cpp:(.text._ZN4llvm27initializeSimpleInlinerPassERNS_12PassRegistryE+0x4e): undefined reference to `llvm::initializeInlineCostAnalysisPass(llvm::PassRegistry&)'
/usr/local/lib/libLLVMipo.a(InlineSimple.cpp.o): In function `(anonymous namespace)::SimpleInliner::getAnalysisUsage(llvm::AnalysisUsage&) const':
/home/nikola/final/llvm.src/lib/Transforms/IPO/InlineSimple.cpp:(.text._ZNK12_GLOBAL__N_113SimpleInliner16getAnalysisUsageERN4llvm13AnalysisUsageE+0xd): undefined reference to `llvm::InlineCostAnalysis::ID'
/usr/local/lib/libLLVMipo.a(InlineSimple.cpp.o): In function `(anonymous namespace)::SimpleInliner::runOnSCC(llvm::CallGraphSCC&)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/InlineSimple.cpp:(.text._ZN12_GLOBAL__N_113SimpleInliner8runOnSCCERN4llvm12CallGraphSCCE+0x1b): undefined reference to `llvm::InlineCostAnalysis::ID'
/home/nikola/final/llvm.src/lib/Transforms/IPO/InlineSimple.cpp:(.text._ZN12_GLOBAL__N_113SimpleInliner8runOnSCCERN4llvm12CallGraphSCCE+0x3c): undefined reference to `llvm::InlineCostAnalysis::ID'
/usr/local/lib/libLLVMipo.a(InlineSimple.cpp.o): In function `(anonymous namespace)::SimpleInliner::getInlineCost(llvm::CallSite)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/InlineSimple.cpp:(.text._ZN12_GLOBAL__N_113SimpleInliner13getInlineCostEN4llvm8CallSiteE+0x20): undefined reference to `llvm::InlineCostAnalysis::getInlineCost(llvm::CallSite, int)'
/usr/local/lib/libLLVMipo.a(InlineSimple.cpp.o):(.data.rel.ro._ZTVN12_GLOBAL__N_113SimpleInlinerE+0x40): undefined reference to `llvm::CallGraphSCCPass::createPrinterPass(llvm::raw_ostream&, std::string const&) const'
/usr/local/lib/libLLVMipo.a(InlineSimple.cpp.o):(.data.rel.ro._ZTVN12_GLOBAL__N_113SimpleInlinerE+0x48): undefined reference to `llvm::CallGraphSCCPass::assignPassManager(llvm::PMStack&, llvm::PassManagerType)'
/usr/local/lib/libLLVMipo.a(PruneEH.cpp.o): In function `llvm::initializePruneEHPass(llvm::PassRegistry&)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/PruneEH.cpp:(.text._ZN4llvm21initializePruneEHPassERNS_12PassRegistryE+0x36): undefined reference to `llvm::initializeCallGraphWrapperPassPass(llvm::PassRegistry&)'
/usr/local/lib/libLLVMipo.a(PruneEH.cpp.o): In function `(anonymous namespace)::PruneEH::runOnSCC(llvm::CallGraphSCC&)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/PruneEH.cpp:(.text._ZN12_GLOBAL__N_17PruneEH8runOnSCCERN4llvm12CallGraphSCCE+0xb7): undefined reference to `llvm::CallGraphWrapperPass::ID'
/home/nikola/final/llvm.src/lib/Transforms/IPO/PruneEH.cpp:(.text._ZN12_GLOBAL__N_17PruneEH8runOnSCCERN4llvm12CallGraphSCCE+0xdc): undefined reference to `llvm::CallGraphWrapperPass::ID'
/usr/local/lib/libLLVMipo.a(PruneEH.cpp.o): In function `(anonymous namespace)::PruneEH::DeleteBasicBlock(llvm::BasicBlock*)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/PruneEH.cpp:(.text._ZN12_GLOBAL__N_17PruneEH16DeleteBasicBlockEPN4llvm10BasicBlockE+0x1f): undefined reference to `llvm::CallGraphWrapperPass::ID'
/home/nikola/final/llvm.src/lib/Transforms/IPO/PruneEH.cpp:(.text._ZN12_GLOBAL__N_17PruneEH16DeleteBasicBlockEPN4llvm10BasicBlockE+0x4c): undefined reference to `llvm::CallGraphWrapperPass::ID'
/home/nikola/final/llvm.src/lib/Transforms/IPO/PruneEH.cpp:(.text._ZN12_GLOBAL__N_17PruneEH16DeleteBasicBlockEPN4llvm10BasicBlockE+0x144): undefined reference to `llvm::CallGraphNode::removeCallEdgeFor(llvm::CallSite)'
/usr/local/lib/libLLVMipo.a(PruneEH.cpp.o):(.data.rel.ro._ZTVN12_GLOBAL__N_17PruneEHE+0x40): undefined reference to `llvm::CallGraphSCCPass::createPrinterPass(llvm::raw_ostream&, std::string const&) const'
/usr/local/lib/libLLVMipo.a(PruneEH.cpp.o):(.data.rel.ro._ZTVN12_GLOBAL__N_17PruneEHE+0x48): undefined reference to `llvm::CallGraphSCCPass::assignPassManager(llvm::PMStack&, llvm::PassManagerType)'
/usr/local/lib/libLLVMipo.a(PruneEH.cpp.o):(.data.rel.ro._ZTVN12_GLOBAL__N_17PruneEHE+0x60): undefined reference to `llvm::CallGraphSCCPass::getAnalysisUsage(llvm::AnalysisUsage&) const'
/usr/local/lib/libLLVMipo.a(ArgumentPromotion.cpp.o): In function `llvm::initializeArgPromotionPass(llvm::PassRegistry&)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/ArgumentPromotion.cpp:(.text._ZN4llvm26initializeArgPromotionPassERNS_12PassRegistryE+0x3e): undefined reference to `llvm::initializeCallGraphWrapperPassPass(llvm::PassRegistry&)'
/usr/local/lib/libLLVMipo.a(ArgumentPromotion.cpp.o): In function `(anonymous namespace)::ArgPromotion::getAnalysisUsage(llvm::AnalysisUsage&) const':
/home/nikola/final/llvm.src/lib/Transforms/IPO/ArgumentPromotion.cpp:(.text._ZNK12_GLOBAL__N_112ArgPromotion16getAnalysisUsageERN4llvm13AnalysisUsageE+0x27): undefined reference to `llvm::CallGraphSCCPass::getAnalysisUsage(llvm::AnalysisUsage&) const'
/usr/local/lib/libLLVMipo.a(ArgumentPromotion.cpp.o): In function `(anonymous namespace)::ArgPromotion::runOnSCC(llvm::CallGraphSCC&)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/ArgumentPromotion.cpp:(.text._ZN12_GLOBAL__N_112ArgPromotion8runOnSCCERN4llvm12CallGraphSCCE+0x9ff): undefined reference to `llvm::CallGraphSCC::ReplaceNode(llvm::CallGraphNode*, llvm::CallGraphNode*)'
/usr/local/lib/libLLVMipo.a(ArgumentPromotion.cpp.o): In function `(anonymous namespace)::ArgPromotion::DoPromotion(llvm::Function*, llvm::SmallPtrSetImpl<llvm::Argument*>&, llvm::SmallPtrSetImpl<llvm::Argument*>&)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/ArgumentPromotion.cpp:(.text._ZN12_GLOBAL__N_112ArgPromotion11DoPromotionEPN4llvm8FunctionERNS1_15SmallPtrSetImplIPNS1_8ArgumentEEES8_+0xf8c): undefined reference to `llvm::CallGraphWrapperPass::ID'
/home/nikola/final/llvm.src/lib/Transforms/IPO/ArgumentPromotion.cpp:(.text._ZN12_GLOBAL__N_112ArgPromotion11DoPromotionEPN4llvm8FunctionERNS1_15SmallPtrSetImplIPNS1_8ArgumentEEES8_+0xfac): undefined reference to `llvm::CallGraphWrapperPass::ID'
/home/nikola/final/llvm.src/lib/Transforms/IPO/ArgumentPromotion.cpp:(.text._ZN12_GLOBAL__N_112ArgPromotion11DoPromotionEPN4llvm8FunctionERNS1_15SmallPtrSetImplIPNS1_8ArgumentEEES8_+0xfc2): undefined reference to `llvm::CallGraph::getOrInsertFunction(llvm::Function const*)'
/home/nikola/final/llvm.src/lib/Transforms/IPO/ArgumentPromotion.cpp:(.text._ZN12_GLOBAL__N_112ArgPromotion11DoPromotionEPN4llvm8FunctionERNS1_15SmallPtrSetImplIPNS1_8ArgumentEEES8_+0x22d6): undefined reference to `llvm::CallGraphNode::replaceCallEdge(llvm::CallSite, llvm::CallSite, llvm::CallGraphNode*)'
/home/nikola/final/llvm.src/lib/Transforms/IPO/ArgumentPromotion.cpp:(.text._ZN12_GLOBAL__N_112ArgPromotion11DoPromotionEPN4llvm8FunctionERNS1_15SmallPtrSetImplIPNS1_8ArgumentEEES8_+0x3106): undefined reference to `llvm::CallGraph::removeFunctionFromModule(llvm::CallGraphNode*)'
/usr/local/lib/libLLVMipo.a(ArgumentPromotion.cpp.o):(.data.rel.ro._ZTVN12_GLOBAL__N_112ArgPromotionE+0x40): undefined reference to `llvm::CallGraphSCCPass::createPrinterPass(llvm::raw_ostream&, std::string const&) const'
/usr/local/lib/libLLVMipo.a(ArgumentPromotion.cpp.o):(.data.rel.ro._ZTVN12_GLOBAL__N_112ArgPromotionE+0x48): undefined reference to `llvm::CallGraphSCCPass::assignPassManager(llvm::PMStack&, llvm::PassManagerType)'
/usr/local/lib/libLLVMipo.a(FunctionAttrs.cpp.o): In function `llvm::initializeFunctionAttrsPass(llvm::PassRegistry&)':
/home/nikola/final/llvm.src/lib/Transforms/IPO/FunctionAttrs.cpp:(.text._ZN4llvm27initializeFunctionAttrsPassERNS_12PassRegistryE+0x3e): undefined reference to `llvm::initializeCallGraphWrapperPassPass(llvm::PassRegistry&)'
/usr/local/lib/libLLVMipo.a(FunctionAttrs.cpp.o): In function `(anonymous namespace)::FunctionAttrs::getAnalysisUsage(llvm::AnalysisUsage&) const':
/home/nikola/final/llvm.src/lib/Transforms/IPO/FunctionAttrs.cpp:(.text._ZNK12_GLOBAL__N_113FunctionAttrs16getAnalysisUsageERN4llvm13AnalysisUsageE+0x3e): undefined reference to `llvm::CallGraphSCCPass::getAnalysisUsage(llvm::AnalysisUsage&) const'
/usr/local/lib/libLLVMipo.a(FunctionAttrs.cpp.o):(.data.rel.ro._ZTVN12_GLOBAL__N_113FunctionAttrsE+0x40): undefined reference to `llvm::CallGraphSCCPass::createPrinterPass(llvm::raw_ostream&, std::string const&) const'
/usr/local/lib/libLLVMipo.a(FunctionAttrs.cpp.o):(.data.rel.ro._ZTVN12_GLOBAL__N_113FunctionAttrsE+0x48): undefined reference to `llvm::CallGraphSCCPass::assignPassManager(llvm::PMStack&, llvm::PassManagerType)'
```

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -183,8 +183,13 @@ sudo pip install pyroute2
 ### Install binary clang
 
 ```
+# FC22
 wget http://llvm.org/releases/3.7.1/clang+llvm-3.7.1-x86_64-fedora22.tar.xz
 sudo tar xf clang+llvm-3.7.1-x86_64-fedora22.tar.xz -C /usr/local --strip 1
+
+# FC23 and FC24
+wget http://llvm.org/releases/3.9.0/clang+llvm-3.9.0-x86_64-fedora23.tar.xz
+sudo tar xf clang+llvm-3.9.0-x86_64-fedora23.tar.xz -C /usr/local --strip 1
 ```
 
 ### Install and compile BCC

